File: frontend/src/features/notification/services/notificationService.ts
import api from '../../../lib/axios';
import type { NotificationItem, PagedResponse } from '../types';

export const notificationService = {
  getNotifications: async (page = 0, size = 20, unreadOnly = true): Promise<PagedResponse<NotificationItem>> => {
    const response = await api.get('/notifications', {
      params: { page, size, unreadOnly },
    });

    return response.data.data;
  },

  markAsRead: async (notificationId: string): Promise<void> => {
    await api.patch(`/notifications/${notificationId}/read`);
  },

  markAllAsRead: async (): Promise<void> => {
    await api.patch('/notifications/read-all');
  },
};

File: backend/src/main/java/com/nchuy099/mini_instagram/notification/service/NotificationService.java
package com.nchuy099.mini_instagram.notification.service;

import com.nchuy099.mini_instagram.common.response.PagedResponse;
import com.nchuy099.mini_instagram.notification.dto.NotificationDTO;
import com.nchuy099.mini_instagram.notification.entity.NotificationType;

import java.util.UUID;

public interface NotificationService {
    PagedResponse<NotificationDTO> getNotifications(int page, int size, boolean unreadOnly);

    void markAsRead(UUID notificationId);
    int markAllAsRead();

    NotificationDTO createNotification(
            NotificationType type,
            UUID actorId,
            UUID recipientId,
            UUID postId,
            UUID commentId
    );
}
